File: backend/src/main/java/com/minegocio/backend/util/ConsultaMapper.java
package com.minegocio.backend.util;

import com.minegocio.backend.dto.consulta.*;
import com.minegocio.backend.model.Consulta;
import com.minegocio.backend.model.Usuario;

public class ConsultaMapper {

    public static ConsultaResponseDto toDto(Consulta c) {
        return ConsultaResponseDto.builder()
                .id(c.getId())
                .idUsuario(c.getUsuario().getId())
                .tipoConsulta(c.getTipoConsulta())
                .descripcion(c.getDescripcion())
                .respuesta(c.getRespuesta())
                .fechaCreacion(c.getFechaCreacion())
                .estado(c.getEstado())
                .build();
    }

    public static Consulta fromCreateDto(ConsultaCreateDto dto, Usuario usuario) {
        return Consulta.builder()
                .usuario(usuario)
                .tipoConsulta(dto.getTipoConsulta())
                .descripcion(dto.getDescripcion())
                .build();
    }
}

File: frontend/src/infrastructure/mappers/consulta.mapper.ts
import type { ConsultaResponseDto } from "../api/dto/FaqDocumentoConsulta.dto";
import type { Consulta } from "../../domain/models/Consulta";

export const mapConsultaDtoToDomain = (dto: ConsultaResponseDto): Consulta => ({
  id: dto.id,
  usuarioId: dto.idUsuario,
  tipo: dto.tipoConsulta,
  descripcion: dto.descripcion,
  respuesta: dto.respuesta ?? undefined,
  fechaCreacion: new Date(dto.fechaCreacion),
  estado: dto.estado,
});
